zipline/utils/calendars/trading_calendar.py

```diff
@@ -889,11 +889,7 @@ def days_at_time(days, t, tz, day_offset=0):
 
 def holidays_at_time(calendar, start, end, time, tz):
     return days_at_time(
-        calendar.holidays(
```

Comment:
:+1: on removing this workaround.  

FWIW, I think the root cause of the failure is actually https://github.com/quantopian/zipline/blob/master/zipline/utils/calendars/trading_calendar.py#L876.  The contract for `days_at_time` is that it's supposed to always return a DatetimeIndex localized to UTC.  We weren't doing that in this case, however, because there were no holidays in the NYSE calendar between 2010-01-03 and 2010-01-10.

I'll fix the empty array case in a separate change.